crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -1,30 +1,51 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::sync::{Arc, Mutex};
 use tokio_util::sync::CancellationToken;
-use tracing::info;
+use tracing::{error, info};
 
 use sui_rest_api::Client;
 
 use crate::types::IndexerResult;
 use crate::{metrics::IndexerMetrics, store::IndexerStore};
+use crate::handlers::TransactionObjectChangesToCommit;
+use crate::store::ObjectChangeToCommit;
 
 const OBJECTS_SNAPSHOT_MAX_CHECKPOINT_LAG: usize = 900;
 const OBJECTS_SNAPSHOT_MIN_CHECKPOINT_LAG: usize = 300;
+// The number of object changes in the buffer before the snapshot processor starts processing them.
+// TODO: placeholder value, need to tune this
```

Comment:
Makes sense, the object updates in this buffer will likely be larger than the typical live objects update but we don't want it to be too large either